packages/iov-jsonrpc/src/parse.ts

```diff
@@ -0,0 +1,90 @@
+import { JsonRpcErrorResponse, JsonRpcRequest, JsonRpcResponse } from "./types";
+
+export function parseJsonRpcId(data: any): number | null {
+  const id = data.id;
+  if (typeof id !== "number") {
+    return null;
+  }
+  return id;
+}
+
+export function parseJsonRpcRequest(data: any): JsonRpcRequest {
+  if (typeof data !== "object") {
+    throw new Error("Data must be an object");
+  }
+
+  if (data.jsonrpc !== "2.0") {
+    throw new Error(`Got unexpected jsonrpc version: ${data.jsonrpc}`);
+  }
+
+  const id = parseJsonRpcId(data);
+  if (id === null) {
+    throw new Error("Invalid id field");
+  }
+
+  const method = data.method;
+  if (typeof method !== "string") {
+    throw new Error("Invalid method field");
+  }
+
+  const params = data.params;
```

Comment:
AFAIK (from experience with tendermint),
Params may be an Array or an object with named arguments.

I personally find the object format easier to work with than the array format.

Reply:
Oh wow, you are right: https://www.jsonrpc.org/specification#parameter_structures "4.2 Parameter Structures". I did not know about this option. Great! Will migrate to it